docs/manual/determinism-checker.tex

```diff
@@ -238,6 +217,48 @@
 % }
 % \end{verbatim}
 
+\sectionAndLabel{Collection invariance}{determinism-invariance}
+The Determinism Checker treats Collection classes as
+invariant with respect to the determinism qualifier. That is, a
+\code{@NonDet Collection} is unrelated by subtyping to an \code{@OrderNonDet Collection}
+which is unrelated to a \code{@Det Collection}.
+Assume for the sake of contradiction that collections are not invariant w.r.t. type qualifiers.
+That would make the following code type-check, even though it
+non-deterministically changes a \code{@Det List}:
+\begin{Verbatim}
+  void test(@Det List<@Det String> dList, @NonDet List<@Det String> nList) {
+    @NonDet List<@Det String> ndList = dList;
+    ndList.addAll(nList);
+  }
+\end{Verbatim}
+
+The above code is unsafe because \<nList> and \<dList> are aliases to
+the same \<List> object and \<nList.addAll(nList)> mutates the \<@Det>
+reference \<dList> non-deterministically.
+
+\sectionAndLabel{Collection aliasing and safe mutation}{determinism-aliasing-mutation}
+It is possible to create \<@NonDet> aliases to \<@OrderNonDet> or \<@Det> collections.
+(For instance, by calling \<next> or \<get>.) Consider the example below:
+\begin{Verbatim}
+  void aliasTest(@OrderNonDet Set<@Det List<@Det String>> set, @NonDet int index, @Det String str) {
+    @NonDet List<@Det String> lst = set.iterator().next();
+    lst.add(index, str);
+  }
+\end{Verbatim}
+
+Variable \<lst> has type \code{@NonDet List<@Det String>} (as a result of set iteration)
+but an alias has type \code{@Det List<@Det String>} (as a member of set).
+The call \<lst.add(...)> is unsafe and must not typecheck. It
+mutates \<lst> non-deterministically thereby violating the determinism
+guarantees provided by the \<@Det> reference.
+
+Our type system prevents the unsafe behavior by prohibiting
+any mutation of collections having types \code{@NonDet Collection<@OrderNonDet E>} or
```

Comment:
If the system prohibits any mutation of collections with that type, why is a method annotation necessary?  Please clarify.